src/test/java/uk/ac/ebi/ena/webin/cli/manifest/processor/metadata/SampleProcessorTest.java

```diff
@@ -12,43 +12,218 @@
 
 import static uk.ac.ebi.ena.webin.cli.manifest.processor.ProcessorTestUtils.createFieldValue;
 
+import java.time.LocalDate;
+import java.util.UUID;
+
 import org.junit.Assert;
 import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.ResourceAccessException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.sun.javafx.binding.StringFormatter;
 
+import uk.ac.ebi.ena.webin.cli.WebinCliMessage;
 import uk.ac.ebi.ena.webin.cli.WebinCliParameters;
 import uk.ac.ebi.ena.webin.cli.WebinCliTestUtils;
 import uk.ac.ebi.ena.webin.cli.manifest.ManifestFieldType;
 import uk.ac.ebi.ena.webin.cli.manifest.ManifestFieldValue;
+import uk.ac.ebi.ena.webin.cli.service.SampleService;
+import uk.ac.ebi.ena.webin.cli.utils.ExceptionUtils;
+import uk.ac.ebi.ena.webin.cli.utils.RemoteServiceUrlHelper;
+import uk.ac.ebi.ena.webin.cli.utils.RetryUtils;
 import uk.ac.ebi.ena.webin.cli.validator.message.ValidationResult;
 import uk.ac.ebi.ena.webin.cli.validator.reference.Sample;
 
 public class
-SampleProcessorTest
-{
+SampleProcessorTest {
     private final WebinCliParameters parameters = WebinCliTestUtils.getTestWebinCliParameters();
+    private final Logger log = LoggerFactory.getLogger(SampleProcessorTest.class);
+    private static final boolean TEST = true;
+
+    @Test
+    public void
+    testCorrect() {
+        SampleProcessor processor = new SampleProcessor(parameters,
+                (Sample sample) -> Assert.assertEquals("SAMEA749881", sample.getBioSampleId()));
 
-    @Test public void 
-    testCorrect()
-    {
-        SampleProcessor processor = new SampleProcessor( parameters,
-            (Sample sample) -> Assert.assertEquals( "SAMEA749881", sample.getBioSampleId() ) );
+        ManifestFieldValue fieldValue = createFieldValue(ManifestFieldType.META, "SAMPLE", "ERS000002");
+        ValidationResult result = new ValidationResult();
+        processor.process(result, fieldValue);
+        Assert.assertTrue(result.isValid());
+        Assert.assertEquals("SAMEA749881", fieldValue.getValue());
+    }
+
+
+    @Test
+    public void
+    testIncorrect() {
+        SampleProcessor processor = new SampleProcessor(parameters, Assert::assertNull);
+        ManifestFieldValue fieldValue = createFieldValue(ManifestFieldType.META, "SAMPLE", "SRP000392");
+        ValidationResult result = new ValidationResult();
+        processor.process(result, fieldValue);
+        Assert.assertFalse(result.isValid());
+        Assert.assertEquals("SRP000392", fieldValue.getValue());
+    }
+
+
+    /**
+     * Test Add and Modify with setSampleUpdate parameter.
```

Comment:
Test add and modify sample submission with -setSampleUpdate option. We change the collection date and check that the sample gets updated. We also test that the BioSample accession is assigned.

```
create testJsonSampleAddAndModify(boolean setSampleUpdate, boolean expectSampleUpdate)
```